From 8868eaba6ab5e616d400276bca92a0bb7481c2c1 Mon Sep 17 00:00:00 2001
From: Rashed Abdel-Tawab <rashed@linux.com>
Date: Wed, 25 Apr 2018 07:27:49 -0700
Subject: [PATCH 1/5] charter: Add some new USB rules

Enforce supporting the same USB version as stock. This is a no-brainer...
We shouldn't ever ship a USB 3.0 device if its only working on USB 2.0
for example.

USB-C devices support more than legacy USB devices, and there is no
reason we shouldn't enforce standards on these. Since they support
host and device mode, along with 5 different power profiles and a
number of video profiles, accessories and other peripherals almost
always depend on normal USB-C profiles.

Change-Id: I3da02a67889806108b9a5ee9c68cb96b18658dbf
---
 device-support-requirements.md | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/device-support-requirements.md b/device-support-requirements.md
index 85cf215..874519b 100644
--- a/device-support-requirements.md
+++ b/device-support-requirements.md
@@ -99,9 +99,13 @@ The use of “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT
 
 ## USB
 
+* All devices with a USB port MUST support the same USB version that they do on stock.
 * All devices with a USB port MUST support file access via MTP.
 * All devices with USB tethering supported on their stock OS MUST support USB tethering.
-* All devices with a USB port & Data SHOULD support USB tethering.
+* All devices with a USB port & mobile data SHOULD support USB tethering.
+* All devices that support USB-OTG on their stock OS MUST support USB-OTG.
+* All USB-C devices SHOULD support all the power profiles supported on their stock OS.
+* All USB-C devices SHOULD support all the video profiles supported on theit stock OS.
 
 ## GPS
 
-- 
2.17.1

